nextcord/components.py

```diff
@@ -230,43 +288,212 @@ class SelectMenu(Component):
     options: List[:class:`SelectOption`]
         A list of options that can be selected in this menu.
     disabled: :class:`bool`
-        Whether the select is disabled or not.
+        Whether the select is disabled or not. Defaults to ``False``.
+    channel_types: List[:class:`ChannelType`]
+        A list of channel types that are allowed to be chosen in this select menu.
     """
 
-    __slots__: Tuple[str, ...] = (
-        "custom_id",
-        "placeholder",
-        "min_values",
-        "max_values",
-        "options",
-        "disabled",
-    )
+    __slots__: Tuple[str, ...] = ("options",)
 
     __repr_info__: ClassVar[Tuple[str, ...]] = __slots__
 
     def __init__(self, data: SelectMenuPayload):
+        super().__init__(data)
         self.type = ComponentType.select
-        self.custom_id: str = data["custom_id"]
-        self.placeholder: Optional[str] = data.get("placeholder")
-        self.min_values: int = data.get("min_values", 1)
-        self.max_values: int = data.get("max_values", 1)
         self.options: List[SelectOption] = [
             SelectOption.from_dict(option) for option in data.get("options", [])
         ]
-        self.disabled: bool = data.get("disabled", False)
 
     def to_dict(self) -> SelectMenuPayload:
-        payload: SelectMenuPayload = {
-            "type": self.type.value,
-            "custom_id": self.custom_id,
-            "min_values": self.min_values,
-            "max_values": self.max_values,
-            "options": [op.to_dict() for op in self.options],
-            "disabled": self.disabled,
-        }
+        payload: SelectMenuPayload = super().to_dict()  # type: ignore
 
-        if self.placeholder:
-            payload["placeholder"] = self.placeholder
+        payload["type"] = self.type.value
```

Comment:
If all select menus have a `type`, it can probably be added to `SelectMenuBase` to reduce repetition.

Reply:
resolved as mentioned in https://canary.discord.com/channels/881118111967883295/881124672056467456/1034564877211148348
"type" cannot be typehinted for the TypedDict.